### PR TITLE
Introduce ThemeProvider for styling across `Features` package

### PR DESCRIPTION
**Pull request by jathushan-r**
### Purpose

This PR addresses an identified issue regarding inconsistent branding and styling when utilizing both Oxygen-UI and Semantic UI React. The implementation introduces a ThemeProvider within common.branding.v1. injecting CSS, it ensures seamless integration of branding elements across the applications.

### Related Issues
- `product-is`: https://github.com/wso2/product-is/issues/20119

### Related PRs
- `identity-apps`: https://github.com/wso2/identity-apps/pull/5849

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/myaccount/src/protected-app.tsx [hunk 1]
```diff
@@ -37,12 +38,13 @@ import axios, { AxiosResponse } from "axios";
 import React, { FunctionComponent, LazyExoticComponent, ReactElement, lazy, useEffect } from "react";
 import { I18nextProvider } from "react-i18next";
 import { useSelector } from "react-redux";
+import { useGetBrandingPreference } from "./api/branding-preferences";
+
```
Comment (JayaShakthi97) [inline apps/myaccount/src/protected-app.tsx:42]: Unnecessary newline
```suggestion
```

File: apps/myaccount/src/protected-app.tsx [hunk 2]
```diff
@@ -184,11 +186,23 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
             .catch((error: string) => {
                 throw new I18nInstanceInitException(error);
             });
+
+
```
Comment (JayaShakthi97) [inline apps/myaccount/src/protected-app.tsx:190]:
```suggestion
```

File: apps/myaccount/src/protected-app.tsx [hunk 3]
```diff
@@ -184,11 +186,23 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
...
     }, [ isAuthenticated ]);
 
+    // Extracting theme preference data using custom hook
```
Comment (JayaShakthi97) [inline apps/myaccount/src/protected-app.tsx:193]: Add full stops at the end of comments. Please check other places as well.
```suggestion
    // Extracting theme preference data using custom hook.
```

File: apps/myaccount/src/protected-app.tsx [hunk 4]
```diff
@@ -184,11 +186,23 @@ export const ProtectedApp: FunctionComponent<AppPropsInterface> = (): ReactEleme
...
+    const { data: themePreference } = useGetBrandingPreference(tenantDomain);
+
+    // Retrieving the application title from the Redux store
+    const appTitle: string = useSelector((state: AppState) => state?.config?.ui?.appTitle);
+
+
```
Comment (JayaShakthi97) [inline apps/myaccount/src/protected-app.tsx:199]:
```suggestion
```

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:19]:
```suggestion
```

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import React, { PropsWithChildren, ReactElement, useMemo } from "react";
+import { Helmet } from "react-helmet";
+import { generateTheme } from "../branding/theme";
+import { ThemeProviderContext, ThemeProviderContextProps } from "../contexts/theme-provider-context";
+import { ThemePreferenceMeta } from "../meta";
+import { BrandingPreferenceAPIResponseInterface } from "../models";
+
+/**
+ * Props interface for the ThemeProvider.
+ */export interface ThemeProviderProps {
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:30]: Check formatting here.
```suggestion
/**
 * Props interface for the ThemeProvider.
 */
export interface ThemeProviderProps {
```

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    // The user's theme preference obtained from the BrandingPreferenceAPI.
+    themePreference: BrandingPreferenceAPIResponseInterface;
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:32]: Use doc comments for props as well. Refer to other places.

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    defaultMode: "light" | "dark";
+
+    // The key used for storing the theme mode preference in local storage.
+    modeStorageKey: string;
+
+    // The title of the application.
+    appTitle?: string;
+}
+
+
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:44]:
```suggestion
```

File: features/common.branding.v1/providers/theme-provider.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+ * @param props - Props for the client.
+ * @returns Theme provider.
+ */
+export const ThemeProvider = (props: PropsWithChildren<ThemeProviderProps>): ReactElement => {
+    const { children,
+        appTitle,
+        themePreference,
+        defaultMode,
+        modeStorageKey
+
```
Comment (JayaShakthi97) [inline features/common.branding.v1/providers/theme-provider.tsx:57]:
```suggestion
```

File: apps/myaccount/src/protected-app.tsx (hunk 1 above)
Comment (jathushan-r) [reply apps/myaccount/src/protected-app.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: apps/myaccount/src/protected-app.tsx (hunk 2 above)
Comment (jathushan-r) [reply apps/myaccount/src/protected-app.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: apps/myaccount/src/protected-app.tsx (hunk 3 above)
Comment (jathushan-r) [reply apps/myaccount/src/protected-app.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 10 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 9 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 8 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 7 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 6 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: features/common.branding.v1/providers/theme-provider.tsx (hunk 5 above)
Comment (jathushan-r) [reply features/common.branding.v1/providers/theme-provider.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: apps/myaccount/src/protected-app.tsx (hunk 4 above)
Comment (jathushan-r) [reply apps/myaccount/src/protected-app.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

File: apps/myaccount/src/protected-app.tsx (hunk 3 above)
Comment (jathushan-r) [reply apps/myaccount/src/protected-app.tsx]: addressed with https://github.com/wso2/identity-apps/pull/6173/commits/651b7a87b8b4c871d6af2f41362a7f8759aa1121

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**